### ISSUE TITLE
【论文复现】经过batch normalization会输出NAN

### ISSUE THREAD

**Issue by thunder95** [PaddlePaddle/Paddle#35518]
经过一定epoch训练后， 输入不是NAN，但是经过BN会输出NAN， 请专家们给个建议，怎么解决？

 ```
weight_attr = paddle.ParamAttr(regularizer=paddle.regularizer.L2Decay(0))
  bias_attr = paddle.ParamAttr(regularizer=paddle.regularizer.L2Decay(0))
  self.bn = nn.BatchNorm2D(num_features=out_channels, epsilon=1e-3, momentum=0.97, weight_attr=weight_attr, bias_attr=bias_attr, use_global_stats=False)
      
```

**Comment by heavengate** [comment]
可能是梯度太大导致权重学习到nan了，可以调小一下learning_rate或者加个gradient clip试下

**Comment by thunder95** [comment]
@heavengate 在optimizer里加了梯度裁剪，仍旧出现这种错误。

```
OptimizerBuilder:
  clip_grad_by_norm: 1.0
  optimizer:
    type: Momentum
    momentum: 0.9
    use_nesterov: true
  regularizer:
    factor: 0.0005
    type: L2

```

**Comment by heavengate** [comment]
把learning rate调小一些试下

**Comment by thunder95** [comment]
@heavengate 学习率调低了也没用

**Comment by thunder95** [comment]
@heavengate 在判断输出nan后，我打印了bn的所有参数（weight, bias, mean, variance），全部是nan， 所以输出也是nan，在bn层的前一个卷积是正常的。

**Comment by heavengate** [comment]
这种情况要么是学习率过大训飞了，要么是梯度过大把权重更新成nan了，可以调小学习率，配合加上grad clip试下，grad clip参考https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/ParamAttr_cn.html#paramattr 
https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/nn/ClipGradByGlobalNorm_cn.html#clipgradbyglobalnorm
  这个在optimizer里设置
